confmap/provider/secretsmanagerprovider/provider.go

```diff
@@ -0,0 +1,60 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package secretsmanagerprovider // import "github.com/open-telemetry/opentelemetry-collector-contrib/confmap/provider/secretsmanagerprovider"
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
+	"go.opentelemetry.io/collector/confmap"
+)
+
+const (
+	schemeName = "secretsmanager"
+)
+
+type provider struct {
+	client *secretsmanager.Client
+}
+
+// New returns a new confmap.Provider that reads the configuration from the given AWS Secrets Manager Name or ARN.
+//
+// This Provider supports "secretsmanager" scheme, and can be called with a selector:
+// `secretsmanager:NAME_OR_ARN`
+func New() confmap.Provider {
```

Comment:
After open-telemetry/opentelemetry-collector/pull/9443 we are passing a `confmap.ProviderSettings` to providers, so I think this should be:

```suggestion
func New(_ confmap.ProviderSettings) confmap.Provider {
```

If you are using this on your vendor distro and this would not work for you we can wait on making this change, but eventually I want to make this change